exporter/containerimage/writer.go

```diff
@@ -392,8 +418,21 @@ func (ic *ImageWriter) rewriteRemoteWithEpoch(ctx context.Context, opts *ImageCo
 		fmt.Sprintf("rewriting layers with source-date-epoch %d (%s)", opts.Epoch.Unix(), opts.Epoch.String()))
 	for i, desc := range remoteDescriptors {
 		i, desc := i, desc
+		info, err := cs.Info(ctx, desc.Digest)
+		if err != nil {
+			return nil, err
+		}
+		diffID := digest.Digest(info.Labels[labels.LabelUncompressed]) // can be empty
+		var immDiffID digest.Digest
+		if baseImg != nil && i < len(baseImg.RootFS.DiffIDs) {
+			immDiffID = baseImg.RootFS.DiffIDs[i]
+			if immDiffID == diffID {
```

Comment:
iiuc this looks for the index match for individual layers. But it should check that full baseImg roofs is a subset in the beginning of the exported image.

Reply:
Added `var divergedFromBase bool` to check this

Reply:
This should be a check before processing. Atm some layers can already be processed before the validation fails.